components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/DatabaseServerImpl.java

```diff
@@ -114,35 +136,72 @@ private OServerConfiguration createConfiguration() {
     System.setProperty(Orient.ORIENTDB_HOME, homeDir.getPath());
 
     OServerConfiguration config = new OServerConfiguration();
-
     // FIXME: Unsure what this is used for, its apparently assigned to xml location, but forcing it here
     config.location = "DYNAMIC-CONFIGURATION";
 
     File databaseDir = applicationDirectories.getWorkDirectory("db");
     config.properties = new OServerEntryConfiguration[]{
-        new OServerEntryConfiguration("server.database.path", databaseDir.getPath())
+        new OServerEntryConfiguration("server.database.path", databaseDir.getPath()),
+        new OServerEntryConfiguration("plugin.dynamic", String.valueOf(dynamicPlugins))
     };
 
-    config.handlers = Lists.newArrayList();
+    config.handlers = new ArrayList<>();
+
+    // configure JMX handler
+    OServerHandlerConfiguration jmxHandler = new OServerHandlerConfiguration();
+    jmxHandler.clazz = OJMXPlugin.class.getName();
+    jmxHandler.parameters = new OServerParameterConfiguration[] {
+        new OServerParameterConfiguration("enabled", "true"),
+        new OServerParameterConfiguration("profilerManaged", "true")
+    };
+    config.handlers.add(jmxHandler);
 
-    config.hooks = Lists.newArrayList();
+    config.hooks = new ArrayList<>();
 
     config.network = new OServerNetworkConfiguration();
     config.network.protocols = Lists.newArrayList(
-        new OServerNetworkProtocolConfiguration("binary", ONetworkProtocolBinary.class.getName())
+        new OServerNetworkProtocolConfiguration("binary", ONetworkProtocolBinary.class.getName()),
+        new OServerNetworkProtocolConfiguration("http", ONetworkProtocolHttpDb.class.getName())
     );
 
-    config.network.listeners = Lists.newArrayList();
+    config.network.listeners = new ArrayList<>();
 
     // HACK: Optionally enable the binary listener
-    if (listenerEnabled) {
-      OServerNetworkListenerConfiguration binaryListener = new OServerNetworkListenerConfiguration();
-      binaryListener.ipAddress = "0.0.0.0";
-      binaryListener.portRange = "2424-2430";
-      binaryListener.protocol = "binary";
-      binaryListener.socket = "default";
-      config.network.listeners.add(binaryListener);
-      log.info("Listener enabled: {}:[{}]", binaryListener.ipAddress, binaryListener.portRange);
+    if (binaryListenerEnabled) {
+      OServerNetworkListenerConfiguration listener = new OServerNetworkListenerConfiguration();
+      listener.ipAddress = "0.0.0.0";
+      listener.portRange = "2424-2430";
+      listener.protocol = "binary";
+      listener.socket = "default";
+      config.network.listeners.add(listener);
+      log.info("Binary listener enabled: {}:[{}]", listener.ipAddress, listener.portRange);
+    }
+
+    // HACK: Optionally enable the http listener
+    if (httpListenerEnabled) {
+      OServerNetworkListenerConfiguration listener = new OServerNetworkListenerConfiguration();
+      listener.ipAddress = "0.0.0.0";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this IP "0.0.0.0" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1313)